hack/select-e2e.sh

```
@@ -0,0 +1,147 @@
+set -eu
+
+CHANGED="${1:?missing changed-files arg}"
+SOURCES_DIR="${2:-packages/core/platform/sources}"
+
+# Anything matching this pattern triggers the full bats suite. Per-app bats
+# (hack/e2e-apps/<name>.bats) are matched BEFORE this so editing one bats
+# file doesn't escalate to the full suite.
+full_suite_pattern='^(packages/library/|packages/core/|api/|cmd/|internal/|hack/[^/]+\.sh$|hack/[^/]+\.bats$|hack/e2e-apps/[^/]+\.sh$|Makefile$|\.github/workflows/(pull-requests|release-e2e)\.yaml$)'
+
+# All known per-app bats files
+all_apps=$(ls hack/e2e-apps/*.bats 2>/dev/null | xargs -n1 basename | sed 's/\.bats$//')
```

Shellcheck (0.11.0)

[warning] Use 'find .. -print0 | xargs -0 ..' or 'find .. -exec .. +' to allow non-alphanumeric filenames.

(SC2011)